oauthproxy.go

```diff
@@ -665,6 +666,44 @@ func (p *OAuthProxy) SignOut(rw http.ResponseWriter, req *http.Request) {
 
 // OAuthStart starts the OAuth2 authentication flow
 func (p *OAuthProxy) OAuthStart(rw http.ResponseWriter, req *http.Request) {
+	extraParams := p.defaultLoginParams()
+	p.overrideLoginParams(extraParams, req)
+	p.doOAuthStart(rw, req, extraParams)
+}
+
+func (p *OAuthProxy) defaultLoginParams() url.Values {
```

Comment:
Wondering if all of this login parameter logic should be a part of the providers package :thinking: 

Counter point I guess is that we don't want to have to reimplement this for every provider :thinking:

Reply:
That was my thought too, it will apply equally to all providers, and it's replacing things like the `prompt` and `acr-values` config options that are not provider specific afaik.

Reply:
I guess it could become part of the core `ProviderData` that is shared amongst all providers 🤔

Reply:
>I guess it could become part of the core ProviderData that is shared amongst all providers 🤔

This is how it works in my latest revision.